doc/decisions/array.md

```diff
@@ -40,11 +38,30 @@ myarray/#5 = value5
   array = #5
 ```
 
+To lookup an array, first do `ksLookup(ks,"/myarray")` on the parent.
+With the last index you get from its metadata `array`, iterate over the children.
+A cascading lookup on every individual child is also needed to make sure that overrides on individual
+elements are respected.
+
+For example:
+
+```
+spec/myarray    # <- contains the specification for the array
+spec/myarray/#  # <- contains the specification for the array elements
+
+dir/myarray/#0  # <- not an array, just an override for user/myarray/#
+
+user/myarray    # <- with metadata array=#0, this would be the array we get
+user/myarray/#0
+
+system/myarray  # <- not found in cascading lookup, as user/myarray exists
```

Comment:
```suggestion
spec:/myarray    # <- contains the specification for the array
spec:/myarray/#  # <- contains the specification for the array elements

dir:/myarray/#0  # <- not an array, just an override for user:/myarray/#

user:/myarray    # <- with metadata array=#0, this would be the array we get
user:/myarray/#0

system:/myarray  # <- not found in cascading lookup, as user:/myarray exists
```

Please use the new notation as far as possible, to avoid additional work in #3447.